pkg/ebpf/events_derived.go

```diff
@@ -116,3 +122,68 @@ func deriveContainerRemoved(t *Tracee) deriveFn {
 		return trace.Event{}, false, nil
 	}
 }
+
+func deriveDetectHookedSyscall(t *Tracee) deriveFn {
+	return func(event trace.Event) (trace.Event, bool, error) {
+		syscallsAdresses, err := getEventArgUlongArrVal(&event, "syscalls_addresses")
+		if err != nil {
+			return trace.Event{}, false, fmt.Errorf("error parsing syscalls_numbers arg: %v", err)
+		}
+		hookedSyscallData := t.ParseHookedAddresses(syscallsAdresses)
+		de := event
+		de.EventID = int(DetectHookedSyscallsEventID)
+		de.EventName = "hooked_syscalls"
+		de.ReturnValue = 0
+		de.Args = []trace.Argument{
+			{ArgMeta: trace.ArgMeta{Name: "hooked_syscalls", Type: "hookedSyscallData[]"}, Value: hookedSyscallData},
+		}
+		de.ArgsNum = 1
+		return de, true, nil
+	}
+}
+
+func (t *Tracee) ParseHookedAddresses(addresses []uint64) []bufferdecoder.HookedSyscallData {
+	hookedSyscallData := make([]bufferdecoder.HookedSyscallData, 0, 0)
+	for idx, syscallsAdress := range addresses {
+		InTextSegment, err := t.kernelSymbols.TextSegmentContains(syscallsAdress)
+		if err != nil {
+			continue
+		}
+		if !InTextSegment {
+			hookingFunction := t.ParseSymbol(syscallsAdress)
+			arch := t.config.OsConfig.GetOSReleaseFieldValue(helpers.OS_ARCH)
+			var syscallNumber int32
+			if strings.Compare(arch, "x86_64") == 0 {
+				if idx > len(syscallsToCheckX86) {
+					continue
+				}
+				syscallNumber = int32(syscallsToCheckX86[idx])
+			} else {
+				if idx > len(syscallsToCheckArm) {
+					continue
+				}
+				syscallNumber = int32(syscallsToCheckArm[idx])
+			}
+			event, found := EventsDefinitions[syscallNumber]
+			var hookedSyscall bufferdecoder.HookedSyscallData
+			if found {
+				hookedSyscall = bufferdecoder.HookedSyscallData{event.Name, hookingFunction.Owner}
```

Comment:
The best practice would be to initialize `hookedSyscall` after the if-else statement, and just initialize the event name according according to if the event was found or not.
You want that only the parts that defer from one another will be in the if-else scope to avoid changes in one branch but not in the other.